test/e2e/service_latency.go

```diff
@@ -85,45 +90,61 @@ var _ = Describe("Service endpoints latency", func() {
 		Logf("50 %%ile: %v", p50)
 		Logf("90 %%ile: %v", p90)
 		Logf("99 %%ile: %v", p99)
+		Logf("Total sample count: %v", len(dSorted))
 
-		if p99 > 4*p50 {
-			failing.Insert("Tail latency is > 4x median latency")
+		if p50 > time.Second*30 {
```

Comment:
30 seconds seems way too long.  I understand that it's what you're seeing empirically, but it seems that the system is broken, and should be fixed, rather than the tests tweaked.